src/main/java/pro/sky/telegrambotanimalshelter/controller/DogController.java

```diff
@@ -0,0 +1,125 @@
+package pro.sky.telegrambotanimalshelter.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pro.sky.telegrambotanimalshelter.exceptions.DogNotFoundException;
+import pro.sky.telegrambotanimalshelter.models.Dog;
+import pro.sky.telegrambotanimalshelter.service.implementation.DogServiceImpl;
+
+import java.util.Collection;
+
+@RestController
+@RequestMapping("dog")
+public class DogController {
+
+    private final DogServiceImpl dogService;
+
+    // Конструктор контроллера, который принимает сервис в качестве зависимости.
+    public DogController(DogServiceImpl dogService) {
```

Comment:
```suggestion
    public DogController(DogService dogService) {
```
зависеть надо от абстракций, а не реализаций